tests/models/test_taskinstance.py

```diff
@@ -2290,6 +2290,26 @@ def test_handle_failure_no_task(self, Stats_incr, dag_maker):
         Stats_incr.assert_any_call('ti_failures')
         Stats_incr.assert_any_call('operator_failures_EmptyOperator')
 
+    def test_handle_failure_task_undefined(self, dag_maker):
+        """
+        When the loaded taskinstance does not use refresh_from_task, the task may be undefined.
+        For example:
+            the DAG file has been deleted before executing _execute_task_callbacks
+        """
+        session = settings.Session()
+        with dag_maker():
+            task = EmptyOperator(task_id="mytask", retries=1)
+        dr = dag_maker.create_dagrun()
+
+        ti: Optional[TI] = (
+            session.query(TI)
+            .filter_by(dag_id=task.dag_id, run_id=dr.run_id, task_id=task.task_id)
+            .one_or_none()
+        )
```

Comment:
```suggestion
        ti: TI = (
            session.query(TI)
            .filter_by(dag_id=task.dag_id, run_id=dr.run_id, task_id=task.task_id)
            .one()
        )
```

This test should fail if there’s not an active ti anyway